src/Form/DomainMenuAccessSettingsForm.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace Drupal\domain_menu_access\Form;
+
+use Drupal\Component\Plugin\Discovery\CachedDiscoveryInterface;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Entity\EntityStorageInterface;
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+class DomainMenuAccessSettingsForm extends ConfigFormBase {
+
+  /**
+   * Block manager
+   */
+  protected $blockManager;
+
+  /**
+   * Menu entities storage
+   */
+  protected $menu;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(ConfigFactoryInterface $config_factory, CachedDiscoveryInterface $block_manager, EntityStorageInterface $menu_storage) {
```

Comment:
Yep, first argument is right, the rest needs clean up
When there's no method in interface just type hint class var to manager instead of interface